terminus-ui/pipes/src/truncate/truncate.pipe.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  Pipe,
+  PipeTransform,
+} from '@angular/core';
+import { isNumber } from '@terminus/ngx-tools';
+
+
+
+/**
+ * Define the accepted string values for the {@link TsTruncateAt} position
+ */
+export type TsTruncatePositionType
+  = 'start'
+  | 'middle'
+  | 'end'
+;
+
+/**
+ * Define the allowed truncation position types Used by {@link TsTruncateAt} position
+ */
+export const allowedTruncationTypes: TsTruncatePositionType[] = [
+  'start',
+  'middle',
+  'end',
+];
+
+
+/**
+ * The truncate at pipe
+ *
+ * @example
+ * {{ 'Here is my string' | tsTruncateAt:7 }} // Outputs: `Here i…`
+ * {{ 'Here is my string' | tsTruncateAt:8:'middle' }} // Outputs: `Here…ing`
+ *
+ * <example-url>https://getterminus.github.io/ui-demos-master/components/pipes</example-url>
+ */
+@Pipe({
+  name: 'tsTruncateAt',
+})
+export class TsTruncateAtPipe implements PipeTransform {
+  transform(value: string, charCount: number = 0, position: TsTruncatePositionType = 'end'): string | undefined {
+    // Check for null values to avoid issues during data-binding
+    if (!value) {
+      return;
+    }
+
+    // Insure the correct type
+    if (!isNumber(charCount)) {
+      console.warn(`${charCount} is not a number.`);
+    }
+
+    if ((value.length < charCount) || (charCount === 0)) {
```

Comment:
We should probably change this to `charCount < 1` instead of just checking for zero. This will protect us against someone passing in a negative number.